### PR TITLE
Add code analysers

### PR DESCRIPTION
**Pull request by neilboyd**
Related:
- https://github.com/NIPOSoftwareBV/nfield-storage/pull/167
- https://github.com/NIPOSoftwareBV/Nfield-Application/pull/569
- https://github.com/NIPOSoftwareBV/nfield-permissions/pull/25
- https://github.com/NIPOSoftwareBV/nfield-identity/pull/371
- https://github.com/NIPOSoftwareBV/Nfield-Capi-Interviewers/pull/449
- https://github.com/NIPOSoftwareBV/Nfield-Manager-Domains/pull/336
- https://github.com/NIPOSoftware/Nfield.Quota/pull/100
- https://github.com/NIPOSoftware/Nfield-SDK/pull/204
- https://github.com/NIPOSoftwareBV/nfield-reporting-events/pull/97
- https://github.com/NIPOSoftwareBV/nfield-nativeengine/pull/83
- https://github.com/NIPOSoftwareBV/nfield-odinparser/pull/49
- https://github.com/NIPOSoftwareBV/nfield-survey-fieldwork-package/pull/27
- https://github.com/NIPOSoftwareBV/nfield-remote-mediator/pull/20
- https://github.com/NIPOSoftwareBV/Nfield-Manager-SurveysAndSampleManagement/pull/1034
- https://github.com/NIPOSoftwareBV/nfield-interviewengine/pull/302
- https://github.com/NIPOSoftwareBV/Nfield-Cati/pull/159
- https://github.com/NIPOSoftwareBV/nfield-source/pull/5182
- https://github.com/NIPOSoftwareBV/project-glu-api/pull/2593
- https://github.com/NIPOSoftwareBV/nfield-interviewing/pull/1259
- https://github.com/NIPOSoftwareBV/nfield-interviewing-contracts/pull/15

Related (no dependencies):
- https://github.com/NIPOSoftwareBV/nfield-tenant-signup/pull/67

[AB#68138](https://niposoftware.visualstudio.com/15ce0e91-931d-4fbf-9169-8c3dde412b54/_workitems/edit/68138)

### CODE REVIEW

File: Nfield.Quota.Tests/QuotaFrameValidatorTests.cs [hunk 1]
```diff
@@ -234,7 +234,7 @@ public void Definitions_CannotHaveInvalidOdinVariableName()
                 "and cannot be empty." +
                 " They can only ​start with​ a letter. First character cannot be '_' or a number." +
                 " Variable definition Id '{0}' with name '{1}' has an invalid Odin Variable Name '{2}'",
-                varId, variableName, invalidOdinVarName);
+                varId, variableName, invalidOdinVarName).Replace("\u200b", "");
```
Comment (carloscoceraperez) [inline Nfield.Quota.Tests/QuotaFrameValidatorTests.cs:237]: Why this change? Looks unrelated to story.

Comment (neilboyd) [reply Nfield.Quota.Tests/QuotaFrameValidatorTests.cs:237]: @joukevandermaas @mvdr do you know anything about this?
I changed `QuotaFrameValidator.cs` to remove the non-printable character, and @krakie changed the test accordingly. However since the test was like this, maybe there was a reason the code was like this?

Comment (neilboyd) [reply Nfield.Quota.Tests/QuotaFrameValidatorTests.cs:237]: #102

Comment (neilboyd) [reply Nfield.Quota.Tests/QuotaFrameValidatorTests.cs]: I removed some invisible spaces from the strings. They don't show up in the diff (because they're invisible, duh!).
I now also undid this change to the test and instead removed some more invisible space in the tests.

Comment (joukevandermaas) [reply Nfield.Quota.Tests/QuotaFrameValidatorTests.cs]: Well, invisible spaces and random characters should not be allowed in odin variable names. So I think that's what the test was testing. I don't really think it should be changed, but you can use escape characters instead if you think that is better.

Comment (neilboyd) [reply Nfield.Quota.Tests/QuotaFrameValidatorTests.cs]: I don't think that's the case. They were just at random places in the message, eg "They can only ​start with​ &lt;invisible space&gt;a letter".
